env/env.cc

```diff
@@ -28,6 +29,21 @@ Status Env::NewLogger(const std::string& fname,
   return NewEnvLogger(fname, this, result);
 }
 
+Status Env::LoadEnv(const std::string& value, Env** result) {
```

Comment:
I think it is possible to call `NewUniqueObject<Env>`. Therefore should we handle this in `LoadEnv()`?

Reply:
There are some potentially memory issues with things like Env* and Comparator* inside the Options object.  These really should be std::shared_ptr objects and not raw pointers.  As it stands now, there is no way to know who is responsible for deleting any of these objects.  If you wrap an Env, does the wrapper delete the inner one?  Who deletes the outer one?

I do not see any places in the code that would return a Unique<Env>.  I would guess that some  of the registration code would return the std::unique_ptr<Env>::get() however.

Reply:
Sure, the Env libraries can handle the life cycle too.